cpp/util/profiler.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2015-2015 the openage authors. See copying.md for legal info.
+
+#include "profiler.h"
+#include <chrono>
+
+namespace openage {
+namespace util {
+
+void Profiler::register_category(Categorie cat) {
```

Comment:
the category registering could be avoided entirely as they all can be just used as default. Using a "activate" and "deactivate" would come in handy, though.